modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/BulkRequestProcessor.java

```diff
@@ -252,4 +283,131 @@ private void errorsCheck(SCIMResponse response) {
         }
     }
 
+    /**
+     * This method is used to process the bulk request content.
+     * This method will replace the bulk id with the created user id.
+     *
+     * @param bulkRequestContent Bulk request content
+     * @param userIDMappings User id bulk id mapping
+     * @param method HTTP method
+     * @throws BadRequestException Bad request exception
+     */
+    private void processBulkRequestContent(BulkRequestContent bulkRequestContent,
+                                           Map<String, String> userIDMappings, String method)
+            throws BadRequestException {
+        try {
+            if (userIDMappings == null
+                    || userIDMappings.isEmpty()
+                    || method.equals(SCIMConstants.OperationalConstants.DELETE)) {
+                return;
+            }
+            // Parse the data field to a JSON object.
+            JSONObject dataJson = new JSONObject(bulkRequestContent.getData());
+            String usersOrMembersKey = getUsersOrMembersKey(bulkRequestContent.getPath());
+            JSONArray usersArray = getUserArray(dataJson, method, usersOrMembersKey);
+
+            if (usersArray != null) {
+                String bulkIdPrefix = SCIMConstants.OperationalConstants.BULK_ID + ":";
+
+                for (int i = 0; i < usersArray.length(); i++) {
+                    JSONObject user = usersArray.getJSONObject(i);
+                    String userValue = user.getString(SCIMConstants.OperationalConstants.VALUE);
+                    if (userValue.startsWith(bulkIdPrefix)) {
```

Comment:
Is there a possibility to be the `userValue` a null? If so better to check that as well.

Reply:
No, when the `user` doesn't have `value` or if the `value` is not a string it will throws a JSONException which is handled.